app/controllers/api/messaging.php

```diff
@@ -0,0 +1,2138 @@
+<?php
+
+use Appwrite\Event\Delete;
+use Appwrite\Event\Messaging;
+use Appwrite\Extend\Exception;
+use Appwrite\Permission;
+use Appwrite\Role;
+use Appwrite\Utopia\Database\Validator\CustomId;
+use Appwrite\Utopia\Database\Validator\Queries\Messages;
+use Appwrite\Utopia\Database\Validator\Queries\Providers;
+use Appwrite\Utopia\Database\Validator\Queries\Subscribers;
+use Appwrite\Utopia\Database\Validator\Queries\Topics;
+use Appwrite\Utopia\Response;
+use Utopia\App;
+use Utopia\Database\Database;
+use Utopia\Database\Document;
+use Utopia\Database\Exception\Duplicate as DuplicateException;
+use Utopia\Database\Helpers\ID;
+use Utopia\Database\Query;
+use Utopia\Database\Validator\Authorization;
+use Utopia\Database\Validator\Datetime as DatetimeValidator;
+use Utopia\Database\Validator\UID;
+use Utopia\Validator\ArrayList;
+use Utopia\Validator\Boolean;
+use Utopia\Validator\JSON;
+use Utopia\Validator\Text;
+use Utopia\Validator\WhiteList;
+
+App::post('/v1/messaging/providers/mailgun')
+    ->desc('Create Mailgun Provider')
+    ->groups(['api', 'messaging'])
+    ->label('audits.event', 'providers.create')
```

Comment:
audits event should be singular:

```suggestion
    ->label('audits.event', 'provider.create')
```